stl/inc/regex

```diff
@@ -574,7 +574,7 @@ private:
                    "could match the specified character sequence.";
         case regex_constants::_Error_parse: // TRANSITION, keeping behavior
             return "regex_error(error_parse)";
-        case regex_constants::error_syntax:
+        case regex_constants::_Error_syntax:
             return "regex_error(error_syntax)";

```

Comment:
No change requested: It would be more consistent to follow the pattern of commenting `// TRANSITION, keeping behavior` here, but the comment on the enumerator definition is sufficient. This isn't worth resetting testing.